src/Components/Forms/DiagnosisAndRecommendations/DiagnosisAndRecommendations.js

```diff
@@ -143,6 +191,156 @@ const DiagnosisAndRecommendations = ({
     }
     return clean;
   };
+
+  const onSubmit = async (data) => {
+    if (isRequiredValidation(data)) {
+      try {
+        const APIsArray = [];
+        const items = qItem.item.map((i) => {
+          const item = {
+            linkId: i.linkId,
+            text: i.text,
+          };
+          switch (i.text) {
+            case 'Finding details':
+              if (data.findingsDetails)
+                item['answer'] = answerType(i.type, data.findingsDetails);
+              break;
+            case 'Diagnosis details':
+              if (data.diagnosisDetails)
+                item['answer'] = answerType(i.type, data.diagnosisDetails);
+              break;
+            case 'Treatment details':
+              if (data.treatmentDetails)
+                item['answer'] = answerType(i.type, data.treatmentDetails);
+              break;
+            case 'Instructions for further treatment':
+              if (data.instructionsForFurtherTreatment)
+                item['answer'] = answerType(
+                  i.type,
+                  data.instructionsForFurtherTreatment,
+                );
+              break;
+            case 'Decision':
+              if (data.decision)
+                item['answer'] = answerType(i.type, data.decision);
+              break;
+            case 'Evacuation way':
+              if (data.evacuationWay)
+                item['answer'] = answerType(i.type, data.evacuationWay);
+              break;
+            case 'Sick leave':
+              if (data.numberOfDays)
+                item['answer'] = answerType(i.type, data.numberOfDays);
+              break;
+            default:
+              break;
+          }
+          return item;
+        });
+        if (Object.keys(normalizedQuestionnaireResponse).length) {
+          APIsArray.push(
+            FHIR('QuestionnaireResponse', 'doWork', {
+              functionName: 'patchQuestionnaireResponse',
+              questionnaireResponseId: normalizedQuestionnaireResponse.id,
+              questionnaireResponseParams: {
+                item: items,
+              },
+            }),
+          );
+        } else {
+          APIsArray.push(
+            FHIR('QuestionnaireResponse', 'doWork', {
+              functionName: 'createQuestionnaireResponse',
+              functionParams: {
+                questionnaireResponse: {
+                  questionnaire: data.questionnaireId,
+                  status: 'completed',
+                  patient: patient.id,
+                  encounter: encounter.id,
+                  authored: moment().format('YYYY-MM-DDTHH:mm:ss[Z]'),
+                  source: patient.id,
+                  item: items,
+                },
+              },
+            }),
+          );
+        }
+        //Updating encounter
+        const cloneEncounter = { ...encounter };
+        cloneEncounter.extensionSecondaryStatus = data.nextStatus;
```

Comment:
the encounter status need to update only when the saving occurs on click a save button and not in the when move between tabs, how is it implemented here?

Reply:
this is not the story for this mission